include/aws/event-stream/event_stream_rpc_server.h

```diff
@@ -0,0 +1,195 @@
+ * continuation options or the program will assert and exit.
+ */
+typedef void(aws_event_stream_rpc_server_on_incoming_stream_fn)(
+    struct aws_event_stream_rpc_server_connection *connection,
+    struct aws_event_stream_rpc_server_continuation_token *token,
+    struct aws_byte_cursor operation_name,
+    struct aws_event_stream_rpc_server_stream_continuation_options *continuation_options,
+    void *user_data);
+
+struct aws_event_stream_rpc_connection_options {
+    aws_event_stream_rpc_server_on_incoming_stream_fn *on_incoming_stream;
+    aws_event_stream_rpc_server_connection_protocol_message_fn *on_connection_protocol_message;
+    void *user_data;
+};
+
+/**
+ * Invoked when a new connection is received on a server listener. You must fill in the fields for connection_options
+ * or the program will assert and exit.
+ *
+ * If error_code is non-zero, an error occurred upon setting up the channel and connection will be NULL. Otherwise,
+ * connection is non-null. If you intend to seat a pointer to connection, you MUST call
+ * aws_event_stream_rpc_server_connection_acquire() and when you're finished with the connection you MUST call
+ * aws_event_stream_server_connection_release().
+ */
+typedef int(aws_event_stream_rpc_server_on_new_connection_fn)(
+    struct aws_event_stream_rpc_server_connection *connection,
+    int error_code,
+    struct aws_event_stream_rpc_connection_options *connection_options,
+    void *user_data);
+
+/**
+ * Invoked when a successfully created connection is shutdown. error_code will indicate the reason for the shutdown.
+ */
+typedef void(aws_event_stream_rpc_server_on_connection_shutdown_fn)(
+    struct aws_event_stream_rpc_server_connection *connection,
+    int error_code,
+    void *user_data);
+
+/**
+ * Invoked whenever a message has been flushed to the channel.
+ */
+typedef void(aws_event_stream_rpc_server_message_flush_fn)(int error_code, void *user_data);
+
+struct aws_server_bootstrap;
+struct aws_event_stream_rpc_server_listener;
+
+/**
+ * (Optional). Invoked when the listener has been successfully shutdown (after the last ref count release).
+ */
+typedef void(aws_event_stream_rpc_server_on_listener_destroy_fn)(
+    struct aws_event_stream_rpc_server_listener *server,
+    void *user_data);
+
+struct aws_event_stream_rpc_server_listener_options {
+    /** host name to use for the listener. This depends on your socket type. */
+    const char *host_name;
+    /** port to use for your listener, assuming for the appropriate socket type. */
+    uint16_t port;
+    const struct aws_socket_options *socket_options;
+    /** optional: tls options for using when setting up your server. */
+    const struct aws_tls_connection_options *tls_options;
+    struct aws_server_bootstrap *bootstrap;
+    aws_event_stream_rpc_server_on_new_connection_fn *on_new_connection;
+    aws_event_stream_rpc_server_on_connection_shutdown_fn *on_connection_shutdown;
```

Comment:
shouldn't `on_connection_shutdown` go in `aws_event_stream_rpc_connection_options` instead of `aws_event_stream_rpc_server_listener_options`?